stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -356,6 +356,11 @@ rdiv!(A::AbstractMatrix{T}, transD::Transpose{<:Any,<:Diagonal{T}}) where {T} =
 (\)(A::Union{QR,QRCompactWY,QRPivoted}, B::Diagonal) =
     invoke(\, Tuple{Union{QR,QRCompactWY,QRPivoted}, AbstractVecOrMat}, A, B)
 
+(/)(U::UpperTriangular, D::Diagonal) = UpperTriangular(parent(U) ./ transpose(D.diag))
```

Comment:
It would be great if you could define the in-place versions `ldiv!` and `rdiv!` instead and then adjust the methods in line 427-429 to use these.

Reply:
Probably just something like
```julia
function rdiv!(U::UpperTriangular, D::Diagonal)
broadcast!(/, parent(U), parent(U), diag(D))
return U
```
although using broadcast here is actually cheating a bit since broadcasting and linear algebra operations have slightly different semantics in corner cases. However, that is an existing issue so we can clean that up in another PR.

Reply:
Hm. The `transpose` is actually also not correct here since it is recursive so you probably want to reshape the vector instead.

Reply:
We'd need the `rdiv!` and `ldiv!` methods anyway and it would be confusing if the non-mutating versions didn't go through the mutating versions. Also, the promotion can be done in the more generic version so it's not that much code. Good idea to use broadcast of the type for that.